### PR TITLE
Add infra for lambda integration tests

### PR DESCRIPTION
**Pull request by rapphil**
**Description:** Add bucket for integration tests of confmap providers in lambda.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cdk_infra/lib/app.ts [hunk 1]
```diff
@@ -10,6 +10,12 @@ const envDefault = {
   account: process.env.CDK_DEFAULT_ACCOUNT,
   region: process.env.CDK_DEFAULT_REGION
 };
+// Lambda CI runs in us-east-1
+const lambdaDefaults = {
+  account: process.env.CDK_DEFAULT_ACCOUNT,
+  region: "us-east-1"
```
Comment (bryan-aguilar) [inline cdk_infra/lib/app.ts]: Is the deployment workflow setup to handle this? Will there be issues with `AWS_REGION` env var that gets set? Is the deployment workflow ready for multi region deployments?

File: cdk_infra/lib/app.ts [hunk 2]
```diff
@@ -10,6 +10,12 @@ const envDefault = {
   account: process.env.CDK_DEFAULT_ACCOUNT,
   region: process.env.CDK_DEFAULT_REGION
 };
+// Lambda CI runs in us-east-1
+const lambdaDefaults = {
+  account: process.env.CDK_DEFAULT_ACCOUNT,
+  region: 'us-east-1'
```
Comment (bryan-aguilar) [inline cdk_infra/lib/app.ts:16]: Is the deployment workflow prepared for multi region deployments? Will there be any issue with the `AWS_REGION` env var being set?

Comment (rapphil) [reply cdk_infra/lib/app.ts:16]: that is why we need to set the region explicitly here.

Comment (rapphil) [reply cdk_infra/lib/app.ts:16]: I think we will need to bootstrap the region though.

Comment (bryan-aguilar) [reply cdk_infra/lib/app.ts:16]: Okay good, I was hoping the explicit region fixed that. Agreed on bootstrapping.

**Review by bryan-aguilar — APPROVED** [review]

**Pull request merged**